src/libraries/LPFeeLibrary.sol

```diff
@@ -25,14 +32,35 @@ library LPFeeLibrary {
         return self & DYNAMIC_FEE_FLAG != 0;
     }
 
+    function isValid(uint24 self, uint24 maxFee) internal pure returns (bool) {
```

Comment:
(optional) seems like this is only used in `validate` -- thinking to just place this logic inside validate to make it easier to read

```
    function validate(uint24 self, uint24 maxFee) internal pure {
        if (self > maxFee) revert FeeTooLarge();
    }
```